### ISSUE TITLE
[IntelliJ][ReportedByUser] Uncaught Exception  java.lang.NullPointerException

### ISSUE THREAD

**Issue by classcial** [microsoft/azure-tools-for-java#3189]
IntelliJ build version: 2018.2.5 IU-182.4892.20
OS: Windows 10
JDK: JetBrains s.r.o 1.8.0_152-release
Plugin version: 3.19.0-2018.2
Additional Info: None
Parent component: 
```
com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog$DialogRootPane[,7,30,824x496,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=449,maximumSize=,minimumSize=,preferredSize=]
```
Error message: 
```
IdeaLoggingEvent[message=null, throwable=java.lang.NullPointerException
	at com.intellij.openapi.wm.impl.InternalDecorator.setTitleActions(InternalDecorator.java:292)
	at com.intellij.openapi.wm.impl.ToolWindowImpl.setTitleActions(ToolWindowImpl.java:309)
	at com.microsoft.intellij.components.ServerExplorerToolWindowFactory.lambda$addToolbarItems$3(ServerExplorerToolWindowFactory.java:374)
	at com.microsoft.azuretools.authmanage.AuthMethodManager.notifySignOutEventListener(AuthMethodManager.java:106)
	at com.microsoft.azuretools.authmanage.AuthMethodManager.signOut(AuthMethodManager.java:162)
	at com.microsoft.azuretools.ijidea.actions.AzureSignInAction.onAzureSignIn(AzureSignInAction.java:114)
	at com.microsoft.azuretools.ijidea.actions.AzureSignInAction.onActionPerformed(AzureSignInAction.java:59)
	at com.microsoft.azuretools.ijidea.utility.AzureAnAction.actionPerformed(AzureAnAction.java:57)

**Comment by wezhang** [comment]
Looks like an auth UX issue

**Comment by Flanker32** [comment]
We didn't remove the tool window listener when close a project, fixed in #3431
